provider/openstack/local_test.go

```diff
@@ -1981,21 +1982,25 @@ func (t *localServerSuite) TestStartInstanceAvailZone(c *gc.C) {
 
 func (t *localServerSuite) TestStartInstanceAvailZoneUnavailable(c *gc.C) {
 	_, err := t.testStartInstanceAvailZone(c, "test-unavailable")
-	c.Assert(err, gc.ErrorMatches, `availability zone "test-unavailable" is unavailable`)
+	// check error logs?
+	//c.Assert(err, gc.ErrorMatches, `availability zone "test-unavailable" is unavailable`)
+	c.Assert(err, gc.ErrorMatches, "failed to start instance in provided availability zone")
```

Comment:
It seems the original error cause - zone being unavailable - is missing now from the error.
I think the underlying error should be annotated so the root cause is not lost.

Reply:
That text is for environs.ErrAvailabilityZoneFailed, which is used to let the provisioner know the error which occurred in StartInstances was a type where a new Availability Zone could work, so try a new one.  Previously, when this was hit: logger.Infof("failed to build instance in availability zone %q", zone), which is what happens now only it moved from the provider to the provisioner.

At least in the OpenStack Provider, the underlying error which triggered trying a new AZ, was not logged explicitly.

TestStartInstanceAvailZoneUnavailable() previously used the StartInstantParams.Placement instead of the new StartInstantParams.AvailabilityZone, which also leads to different errors.